dev/dev-env-setup.sh

```diff
@@ -340,11 +357,30 @@ set_pre_commit_and_git_signoff() {
 # Execute mandatory setups with strict error handling
 set +xv && set -e
 # Mandatory setups
-check_and_install_pyenv
-check_and_install_min_py_version
-create_virtualenv
-install_mlflow_and_dependencies
-set_pre_commit_and_git_signoff
+if [[ "$PROGRESS" -eq "0" ]]; then
```

Comment:
honestly I wasn't happy with the lengthy if conditions. Original plan is to do case switch + fall through to make the code cleaner. However, ;& and ;;& were introduced only in bash 4.0. Did some research and found some MacOS versions still only offers bash 3.x - think it's too much to prompt the users to upgrade their bash since the setup script is supposed to make the onboarding as easy as possible. Hence went for this approach that should work for all bash versions

```
case $PROGRESS in
  0)
    check_and_install_pyenv
    save_progress 1
    ;&
  1)
    check_and_install_min_py_version
    save_progress 2
    ;&
  2)
    create_virtualenv
    save_progress 3
    ;&
  3)
    install_mlflow_and_dependencies
    save_progress 4
    ;&
  4)
    set_pre_commit_and_git_signoff
    save_progress 5
    ;&
  5)
    # Clear progress file if all mandatory steps are executed successfully
    rm $PROGRESS_FILE
    ;;
esac
```